jupyter-docker/jupyter_localize_extension.py

```diff
@@ -6,34 +6,76 @@
 from tornado.web import HTTPError
 from notebook.base.handlers import IPythonHandler
 from notebook.utils import url_path_join
+from datauri import DataURI
 
 class LocalizeHandler(IPythonHandler):
-  def sanitize(self, pathstr):
-    """Expands to absolute paths, makes intermediate dirs, and quotes to remove any shell naughtiness.
-    This doesn't need to be a coroutine as it's an inline function, not something Future-y."""
-    #expanduser behaves fine with gs:// urls, thankfully
-    expanded = os.path.expanduser(pathstr)
-    if not pathstr.startswith("gs://"):
+  def _sanitize(self, pathstr):
+    """Sanitizes paths. Handles local paths, gs: URIs, and data: URIs."""
+    # return gs or data uris as is
+    if pathstr.startswith("gs:") or pathstr.startswith("data:"):
+      return pathstr
+    # expand user directories and make intermediate directories
+    else:
+      expanded = os.path.expanduser(pathstr)
       try:
         os.makedirs(os.path.dirname(expanded))
       except OSError: #thrown if dirs already exist
         pass
-    return pipes.quote(expanded)
+      return expanded
+
+  def _handle_gcs_uri(self, locout, source, dest):
```

Comment:
Can we be more specific about what `handle` means?

Reply:
Maybe `_localize_gcs_uri` would be clearer?